packages/hd-cache/src/index.ts

```diff
@@ -201,16 +202,24 @@ export class HDCache {
   }
 
   getNextReceivingPublicKeyInfo(): PublicKeyInfo {
-    const info: PublicKeyInfo = this.receivingKeys.find(
+    const info: PublicKeyInfo | undefined = this.receivingKeys.find(
       (key) => key.historyTxCount === 0
-    )!;
+    );
+    /* c8 ignore next 3 */
+    if (!info) {
+      throw new Error("Impossible: can not find receivingKey");
+    }
```

Comment:
i find there are many of if(!condition) { throw new impossible error }, would you please refactor to assertion style like this

```ts
function assertsNonNil(value: unknown): asserts value is NonNullable<T> {
  if (value === undefined || value === null) throw new Error("...")
}
```